controller/src/main/java/org/jboss/as/controller/operations/global/ReadAttributeHandler.java

```diff
@@ -281,9 +281,19 @@ static void resolveAttribute(OperationContext context, AttributeDefinition attri
             final ModelNode result = subModel.get(attribute.getName());
             context.getResult().set(result);
         } else if (defaults && attribute.getDefaultValue() != null) {
-            // No defined value in the model. See if we should reply with a default from the metadata,
-            // reply with undefined, or fail because it's a non-existent attribute name
-            context.getResult().set(attribute.getDefaultValue());
+            if(!attribute.hasAlternative(subModel)) {
```

Comment:
I don't think we want to back port this at this point. It's not in WildFly Core currently and I'm not sure it will make it into EAP 7.1. @bstansberry can make the final call, but it's too late to change this behavior IMO. It's been like this for some time now.